### DISCUSSION TITLE
Using composables within a composable function

### DISCUSSION THREAD

**Discussion by richgcook** [nuxt/nuxt#22893] [Questions]
I'm a little confused, due to auto importing within script setup how to use a composable within a composable?

I want to use useRoute, useNuxtApp, useSanityQuery and useHead in a composable to save me including the same 20/30 lines of code for every page template but I'm confused exactly how to do this. Can anyone help?

```
export const usePageSetup = (query, params) => {

	const route = useRoute()
  	const { $seoQuery, $imageQuery, $richTextQuery } = useNuxtApp()

	const defaultParams = {
		seoQuery: $seoQuery, 
		imageQuery: $imageQuery, 
		richTextQuery: $richTextQuery
	}

	const { data } = await useSanityQuery(query, { ...defaultParams, ...params })

	const { title, description, image, bodyClass } = useMetaBodyHelpers(data.value)

	const pageTitle = title.value
	const pageDescription = description.value
	const pageImage = image.value
	const pageClass = bodyClass.value

	useHead({
		title: pageTitle,
		meta: useCreateSeoMeta({ pageTitle, pageDescription, pageImage }),
		bodyAttrs: {
			class: pageClass
		}
	})

	return {
		data,
		pageTitle,
		pageDescription,
		pageImage,
		pageClass
	}

}
```

**Comment by xjccc** [comment]
can use a nuxt plugin to resolve ?

**Comment by TheAlexLichter** [comment]
Yes, that works fine and is actually desired 👍🏻

Only thing missing might be the `async` in front of the arrow fn because you use `await`.

Please provide a reproduction (e.g. StackBlitz) if something doesn't work as planned
